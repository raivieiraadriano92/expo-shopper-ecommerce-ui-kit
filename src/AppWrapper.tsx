import { FunctionComponent } from 'react'

import {
  useFonts,
  Poppins_900Black,
  Poppins_700Bold,
  Poppins_800ExtraBold,
  Poppins_200ExtraLight,
  Poppins_300Light,
  Poppins_500Medium,
  Poppins_400Regular,
  Poppins_600SemiBold,
  Poppins_100Thin
} from '@expo-google-fonts/poppins'
import { NavigationContainer } from '@react-navigation/native'
import AppLoading from 'expo-app-loading'
import { NativeBaseProvider } from 'native-base'

import { theme } from './theme'

export const AppWrapper: FunctionComponent = ({ children }) => {
  const [fontsLoaded] = useFonts({
    Poppins_900Black,
    Poppins_700Bold,
    Poppins_800ExtraBold,
    Poppins_200ExtraLight,
    Poppins_300Light,
    Poppins_500Medium,
    Poppins_400Regular,
    Poppins_600SemiBold,
    Poppins_100Thin
  })

  return !fontsLoaded ? (
    <AppLoading />
  ) : (
    <NativeBaseProvider
      config={{
        enableRem: true,
        suppressColorAccessibilityWarning: true
      }}
      initialWindowMetrics={{
        frame: { x: 0, y: 0, width: 0, height: 0 },
        insets: { top: 0, left: 0, right: 0, bottom: 0 }
      }}
      {...{ theme }}>
      <NavigationContainer>{children}</NavigationContainer>
    </NativeBaseProvider>
  )
}
