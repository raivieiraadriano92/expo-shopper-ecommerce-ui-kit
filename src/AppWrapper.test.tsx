import { render, waitFor } from '@testing-library/react-native'
import { Text } from 'native-base'

import { AppWrapper } from './AppWrapper'

describe('<AppWrapper />', () => {
  it('renders correctly', async () => {
    const { findByText, toJSON } = render(
      <AppWrapper>
        <Text>testing...</Text>
      </AppWrapper>
    )

    await waitFor(() => findByText('testing...'))

    expect(toJSON()).toMatchSnapshot()
  })
})
