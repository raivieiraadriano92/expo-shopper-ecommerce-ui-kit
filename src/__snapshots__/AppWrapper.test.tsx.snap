// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppWrapper /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
  testID="AppWrapper"
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  />
</View>
`;
